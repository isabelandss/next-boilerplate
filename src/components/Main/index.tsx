const Main = () => (
  <main>
    <h1>react avanÃ§ado</h1>
  </main>
)

export default Main
