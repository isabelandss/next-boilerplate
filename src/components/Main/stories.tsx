import { Meta, Story } from '@storybook/react/types-6-0'
import { text, withKnobs } from '@storybook/addon-knobs'

import Main from '.'

export default {
  title: 'Main',
  component: Main,
  decorators: [withKnobs],
} as Meta

export const Basic: Story = () => (
  <Main
    title={text('title', 'React AvanÃ§ado')}
    description={text(
      'description',
      'TypeScript, React, Next e Styles Components',
    )}
  />
)
