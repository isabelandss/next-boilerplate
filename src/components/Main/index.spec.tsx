import { render, screen } from '@testing-library/react'

import Main from '.'

describe('<Main />', () => {
  test('should match snapshot', () => {
    const { container } = render(<Main />)
    expect(container.firstChild).toMatchSnapshot()
  })

  test('should render the heading', () => {
    render(<Main />)

    expect(
      screen.getByRole('heading', { name: /react avan√ßado/i }),
    ).toBeInTheDocument()
  })
})
