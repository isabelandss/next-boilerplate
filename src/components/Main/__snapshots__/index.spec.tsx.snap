// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should match snapshot 1`] = `
<main
  class="indexstyles__Wrapper-sc-1y7lmfi-0 eRpYSd"
>
  <img
    alt="Imagem de um átomo e React Avançado escrito ao lado"
    class="indexstyles__Logo-sc-1y7lmfi-1 lbvvUy"
    src="/img/logo.svg"
  />
  <h1
    class="indexstyles__Title-sc-1y7lmfi-2 rYetY"
  >
    React Avançado
  </h1>
  <h2
    class="indexstyles__Description-sc-1y7lmfi-3 dHjBZn"
  >
    TypeScript, React, Next e Styles Components
  </h2>
  <img
    alt="Um desenvolvedor de frente para uma tela com código"
    class="indexstyles__Illustration-sc-1y7lmfi-4 fLLOqO"
    src="/img/hero-illustration.svg"
  />
</main>
`
